extensions/VisualEditor/wikia/modules/ve/ui/dialogs/ve.ui.WikiaInfoboxDialog.js

```diff
@@ -0,0 +1,210 @@
+/*
+ * VisualEditor user interface WikiaInfoboxDialog class.
+ */
+
+/**
+ * Dialog for inserting and editing Wikia Infobox templates.
+ *
+ * @class
+ * @extends ve.ui.NodeDialog
+ *
+ * @constructor
+ * @param {Object} [config] Configuration options
+ */
+ve.ui.WikiaInfoboxDialog = function VeUiWikiaInfoboxDialog( config ) {
+	// Parent constructor
+	ve.ui.WikiaInfoboxDialog.super.call( this, config );
+};
+
+/* Inheritance */
+
+OO.inheritClass( ve.ui.WikiaInfoboxDialog, ve.ui.NodeDialog );
+
+/* Static Properties */
+
+ve.ui.WikiaInfoboxDialog.static.name = 'wikiaInfobox';
+
+ve.ui.WikiaInfoboxDialog.static.modelClasses = [ ve.dm.WikiaInfoboxTransclusionBlockNode ];
+
+ve.ui.WikiaInfoboxDialog.static.actions = [
+	{
+		action: 'apply',
+		label: OO.ui.deferMsg( 'visualeditor-dialog-action-apply' ),
+		flags: 'primary'
+	},
+	{
+		label: OO.ui.deferMsg( 'visualeditor-dialog-action-cancel' ),
+		flags: 'safe'
+	}
+];
+
+/* Methods */
+
+/**
+ * @inheritdoc
+ */
+ve.ui.WikiaInfoboxDialog.prototype.getActionProcess = function ( action ) {
+	if ( action === 'apply' ) {
+		return new OO.ui.Process( function () {
+			var surfaceModel = this.getFragment().getSurface();
+
+			//check to be sure, user clicked on infobox so it always has to be an instance of MWTransclusionNode
+			if ( this.selectedNode instanceof ve.dm.MWTransclusionNode ) {
```

Comment:
Why is this check needed? If there is a situation that this method (ve.ui.WikiaInfoboxDialog.prototype.getActionProcess) is called and selectNode is different than ve.dm.MWTransclusionNode, well, then we might have a bigger problem.